code/plotting.py

```diff
@@ -377,15 +377,41 @@ def do_plot(
 
 if plotting['all_methods_separate']:
 
-    plot_input = np.array([
-        [variables_to_plot['p_values_expected_rate'], variables_to_plot['best_fit_expected_rate'], "Rate", "black", "solid"],
-        [variables_to_plot['p_values_expected_histo'], variables_to_plot['best_fit_expected_histo'], "Histo", "limegreen", "solid"],
-        [variables_to_plot['p_values_expected_histo_kin'], variables_to_plot['best_fit_expected_histo_kin'], "Histo-Kin", "limegreen", "dashed"],
-        [variables_to_plot['p_values_expected_sally'], variables_to_plot['best_fit_expected_sally'], "SALLY", "C0", "solid"],
-        [variables_to_plot['p_values_expected_sally_kin'], variables_to_plot['best_fit_expected_sally_kin'], "SALLY-Kin", "C0", "dashed"],
-        [variables_to_plot['p_values_expected_alices'], variables_to_plot['best_fit_expected_alices'], "ALICES", "red", "solid"],
-        [variables_to_plot['p_values_expected_alices_kin'], variables_to_plot['best_fit_expected_alices_kin'], "ALICES-Kin", "red","dashed"],
-    ])
+    p_values_expected = 'p_values_expected_'+plotting_method
+    best_fit_expected = 'best_fit_expected_'+plotting_method
+    p_values_expected_kin = 'p_values_expected_'+plotting_method+'_kin'
+    best_fit_expected_kin = 'best_fit_expected_'+plotting_method+'_kin'
```

Comment:
Please, use the [f-string formatting](https://docs.python.org/3/tutorial/inputoutput.html#formatted-string-literals) to build these strings (faster and more readable).

```python3
p_values_expected = f"p_values_expected_{plotting_method}"
best_fit_expected = f"best_fit_expected_{plotting_method}"
p_values_expected_kin = f"p_values_expected_{plotting_method}_kin"
best_fit_expected_kin = f"best_fit_expected_{plotting_method}_kin"
```